### PR TITLE
feat: added appInsights to CSR

### PR DESCRIPTION
**Pull request by naren-bellala**
#### 📲 What

A description of the change.

[AB#1729](https://amido-dev.visualstudio.com/73884c9a-a68f-4f67-b2b5-b588c2eb8492/_workitems/edit/1729)

#### 🤔 Why
		
CSR Logging parity with SSR
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshot 2020-05-21 at 16 10 34

Screenshot 2020-05-21 at 16 13 09

Screenshot 2020-05-21 at 16 16 21
		 
#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [x] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [x] Passing any exploratory testing?
- [x] Rebased/merged with latest changes from development and re-tested?
- [x] Meeting the Coding Standards?

### CODE REVIEW

File: packages/scaffolding-cli/templates/src/csr/src/services/apiCaller.ts [hunk 1]
```diff
@@ -19,6 +20,8 @@ export default function apiCaller<T>(
             return res.data.results
         })
         .catch(err => {
+            // appInsights: track error event
+            trackError(err)
```
Comment (estherthetesteramido) [inline packages/scaffolding-cli/templates/src/csr/src/services/apiCaller.ts:24]: How do we split the events in app insights? is there custom telemetry params that we can send to determine the client?

Comment (naren-bellala) [reply packages/scaffolding-cli/templates/src/csr/src/services/apiCaller.ts:24]: Custom telemetry meta data like project name and version are automatically logged on each tracking call. "@microsoft/applicationinsights-react-js" lib automatically understands React SPA pages.

Comment (naren-bellala) [reply packages/scaffolding-cli/templates/src/csr/src/services/apiCaller.ts:24]: I will add the custom trace function now. Good shout.

Comment (naren-bellala) [reply packages/scaffolding-cli/templates/src/csr/src/services/apiCaller.ts:24]: Added

**Comment by naren-bellala** [comment]
> Can we have evidence of this working in AppInsights?

Attached in the PR

**Pull request merged**